lighthouse-core/gather/gatherers/inspector-issues.js

```diff
@@ -38,20 +38,27 @@ class Issues extends Gatherer {
 
   /**
    * @param {LH.Gatherer.PassContext} passContext
-   * @return {Promise<LH.Artifacts['Issues']>}
+   * @return {Promise<LH.Artifacts['InspectorIssues']>}
    */
   async afterPass(passContext) {
     const driver = passContext.driver;
 
     driver.off('Audits.issueAdded', this._onIssueAdded);
     await driver.sendCommand('Audits.disable');
-    return this._issues.map(function(event) {
-      return {
-        code: event.issue.code,
-        details: event.issue.details,
-      };
-    });
+    /** @type {Array<LH.Crdp.Audits.MixedContentIssueDetails>} */
+    const mixedContentIssues = [];
```

Comment:
ideas for making this cleaner, easily extensible later on:

1. define the artifact obj first.
```
const artifacts = {
  mixedContentIssues: [],
}
```

2. use a for/of loop on this._issues

3. naming–instead of `mixedContentIssues`, what about `mixedContent` or even `MixedContent`?

Reply:
Just to make sure I'm clear, you mean use the for/of loop in place of the filter/forEach chaining that I have now? Then the inside would look something like w/ your suggestions:
```
if (issue.code === 'MixedContentIssue') {
  artifact.MixedContent.push(issue.details.mixedContentIssueDetails);
}
```
Am I getting this right?

Reply:
yes